src/components/SchoolsLookUp/SchoolsLookUp.js

```diff
@@ -33,32 +37,59 @@ class SchoolsLookUp extends Component {
     let lookup;
     if (selectedEstablishment && selectedEstablishment.id) {
       lookup = HIDE_LOOKUP;
-    } else if (establishmentNameValue) {
+    } else if (establishmentNameValue || this.hasError(props)) {
       lookup = SHOW_MANUAL_LOOKUP;
     } else {
       lookup = SHOW_EDCO_LOOKUP;
     }
     this.state = {
       options: [],
+      query: '',
+      isSearching: false,
       lookup,
     };
-    this.renderMenuItemChildren = SchoolsLookUp.renderMenuItemChildren;
+
     this.handleChange = this.handleChange.bind(this);
     this.handleSearch = this.handleSearch.bind(this);
     this.handleLookup = this.handleLookup.bind(this);
     this.handleManual = this.handleManual.bind(this);
     this.handleBlur = this.handleBlur.bind(this);
+    this.renderMenu = this.renderMenu.bind(this);
     this.renderSingleInput = this.renderSingleInput.bind(this);
     this.renderEstablishmentDetails = this.renderEstablishmentDetails.bind(this);
   }
 
   /**
+   * Component lifecycle event triggered whenever props change
+   * @param {object} props
+   */
+  componentWillReceiveProps(nextProps) {
+    if (this.hasError(nextProps)) {
+      this.handleLookup(SHOW_MANUAL_LOOKUP);
+    }
+  }
+
+  /**
+   * Check whether any field has error or all are valid
+   * @param {object} props
+   * @return {boolean}
```

Comment:
I believe right now this will actually return a string or `undefined`? We rely later on it being cast to bool but that's not this fn's return type.

I think it would be good to explain this and change the type hint, or maybe better to just cast to boolean explicitly inside this fn.

Reply:
fixed